src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/TargetsTest.java

```diff
@@ -510,4 +510,194 @@ public void testNoOCPTargetHotplugFlagSupportedOS() {
+        Assert.assertTrue(report.getRecommendedTargetsIMS().stream().anyMatch(target -> target.toLowerCase().contains("None".toLowerCase())));
+        Assert.assertFalse(report.getRecommendedTargetsIMS().stream().anyMatch(target -> target.toLowerCase().contains("OpenJDK".toLowerCase())));
+    }
+
+    @Test
+    public void testGivenUbuntuOracleJDK8AndWeblogic_ThenNoOpenJDKRecommendedTarget() {
+        Map<String, Object> facts = new HashMap<>();
+        // always add a String fact with the name of the agenda group defined in the DRL file (e.g. "SourceCosts")
+        facts.put("agendaGroup", "Targets");
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+        workloadInventoryReportModel.setOsFamily("Ubuntu");
+        workloadInventoryReportModel.addWorkload("Oracle JDK 8");
+
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+
+        Assert.assertEquals(2, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener,
+                "AgendaFocusForTest",
+                "Target_None"
+        );
+
+        List<WorkloadInventoryReportModel> reports = Utils.extractModels(GET_OBJECTS_KEY, results, WorkloadInventoryReportModel.class);
+
+        // just one report has to be created
+        Assert.assertEquals(1, reports.size());
+        WorkloadInventoryReportModel report = reports.get(0);
+        Assert.assertNotNull(report.getRecommendedTargetsIMS());
+        Assert.assertEquals(1, report.getRecommendedTargetsIMS().size());
+        Assert.assertTrue(report.getRecommendedTargetsIMS().stream().anyMatch(target -> target.toLowerCase().contains("None".toLowerCase())));
+        Assert.assertFalse(report.getRecommendedTargetsIMS().stream().anyMatch(target -> target.toLowerCase().contains("OpenJDK".toLowerCase())));
+    }
+
+    @Test
+    public void testGivenRHELOracleJDK8_ThenOpenJDKRecommendedTarget() {
+        Map<String, Object> facts = new HashMap<>();
+        // always add a String fact with the name of the agenda group defined in the DRL file (e.g. "SourceCosts")
+        facts.put("agendaGroup", "Targets");
+
+        WorkloadInventoryReportModel workloadInventoryReportModel = new WorkloadInventoryReportModel();
+        workloadInventoryReportModel.setOsFamily("RHEL");
+        workloadInventoryReportModel.addWorkload("Oracle JDK 8");
+
+        facts.put("workloadInventoryReportModel",workloadInventoryReportModel);
+
+        Map<String, Object> results = createAndExecuteCommandsAndGetResults(facts);
+
+        Assert.assertEquals(3, results.get(NUMBER_OF_FIRED_RULE_KEY));
+        Utils.verifyRulesFiredNames(this.agendaEventListener,
+                "AgendaFocusForTest",
+                "Target_None",
+                "Target_OpenJDK"
```

Comment:
I don't think we should have Target_None rule firing with another target, ie the new  Target_OpenJDK rule. It might be worth checking with Miguel what he wants wrt this but my instinct is that we should modify Target_None so that it doesn't fire if Target_OpenJDK does.

Reply:
@m-brophy Thanks for your PR https://github.com/project-xavier/xavier-analytics/pull/64/commits/d333502447a610d151830617b78d03ff678356e9. Your instinct was absolutely right and I think you answered yourself with your PR. This comment should be solved by last commit (the one that comes from your PR)